src/operator/nn/dropout-inl.h

```diff
@@ -82,7 +82,7 @@ class DropoutOp {
   static void BernoulliGenerate(common::random::RandGenerator<cpu, DType> gen,
                                 int n, double p, int* r) {
     typename RandGenerator<xpu, DType>::Impl genImpl(&gen, 1);
-    const int seed = 17 + genImpl.rand() % 4096;  // NOLINT(runtime/threadsafe_fn)
+    const int seed = 17 + abs(genImpl.rand() % 4096);
     const int nthr = engine::OpenMP::Get()->GetRecommendedOMPThreadCount();
 #pragma omp parallel num_threads(nthr)
```

Comment:
I think it will be better to change this parallel approach to "#pragma omp paralle for" so it will be more readable than OMP section parallel.
Do you mind to change it?

Reply:
@pengzhao-intel Thanks for your feedback. I don't see why "#pragma omp parallel for" would be more readable in this case as there is no for loop involved. The job for each thread is defined based on its thread number. Introducing a for loop with "#pragma omp parallel for" would be redundant. Please correct me if I've misunderstood it.

Reply:
@apeforest vslNewStream expects a non-negative integer as seed by definition. genImpl.rand() generates negative numbers too, which sometimes leads to negative seed values being passed to vslNewStream. abs() helps keep that in check.

Reply:
Thanks for the explanation. In this case I would suggest we add an assert here to make sure seed is nonnegative

Reply:
Added a check to make sure seed is non-negative.